packages/suite/src/utils/suite/date.ts

```diff
@@ -84,14 +85,18 @@ export const calcTicksFromData = (data: { time: number }[]) => {
 };
 
 /**
- * Returns Blockbook-safe current unix timestamp (current time - 3 mins).
+ * Returns Blockbook-safe current unix timestamp (time - 3 mins).
  * Little workaround for Blockbook as it doesn't return rates data for too recent timestamps.
  *
  * @returns
  */
-export const getBlockbookSafeTime = () => {
-    const timestamp = getUnixTime(new Date());
-    return timestamp - 180; // current time - 3 mins
+export const getBlockbookSafeTime = (ts?: number) => {
+    const currentTimestamp = getUnixTime(new Date());
+    if (ts && differenceInMinutes(currentTimestamp * 1000, ts * 1000) > 3) {
+        // timestamp is older than 3 mins, no adjustment needed
+        return ts;
+    }
+    return currentTimestamp - 180;
```

Comment:
A shorter alternative could be this. I like that it doesn't use the `differenceInMinutes` function which isn't particularly clear on first sight. I'm not sure if I like the ternary though. Feel free to take/edit/discard this as you wish and resolve this yourself. .)

```suggestion
export const getBlockbookSafeTime = (ts?: number) => {
    const threeMinutesAgo = getUnixTime(new Date()) - 180;
    return ts !== undefined ? min(ts, threeMinutesAgo) : threeMinutesAgo;
}
```

Reply:
I find both equally hard to read 😁  I'll leave it as it is if you don't mind

Reply:
no prob. Resolving.